appbuilder/services/npm-service.ts

```diff
@@ -0,0 +1,233 @@
+	}
+
+	private installTypingsForDependency(projectDir: string, dependency: string): IFuture<void> {
+		return this.npmInstall(projectDir, `${NpmService.TYPES_DIRECTORY}${dependency}`, null, ["--save-dev", "--save-exact"]);
+	}
+
+	private generateReferencesFile(projectDir: string): IFuture<void> {
+		return (() => {
+			let packageJsonContent = this.getPackageJsonContent(projectDir).wait();
+
+			let pathToReferenceFile = this.getPathToReferencesFile(projectDir),
+				lines: string[] = [];
+
+			if (packageJsonContent && packageJsonContent.dependencies && packageJsonContent.dependencies[constants.TNS_CORE_MODULES]) {
+				let relativePathToTnsCoreModulesDts = `./${constants.NODE_MODULES_DIR_NAME}/${constants.TNS_CORE_MODULES}/${NpmService.TNS_CORE_MODULES_DEFINITION_FILE_NAME}`;
+
+				if (this.$fs.exists(path.join(projectDir, relativePathToTnsCoreModulesDts)).wait()) {
+					lines.push(this.getReferenceLine(relativePathToTnsCoreModulesDts));
+				}
+			}
+
+			if (packageJsonContent && packageJsonContent.devDependencies) {
+				_(packageJsonContent.devDependencies)
+					.keys()
+					.each(devDependency => {
+						if (this.isFromTypesRepo(devDependency)) {
+							let nodeModulesDirectory = path.join(projectDir, constants.NODE_MODULES_DIR_NAME);
+							let definitionFiles = this.$fs.enumerateFilesInDirectorySync(path.join(nodeModulesDirectory, devDependency),
+								(file, stat) => _.endsWith(file, constants.FileExtensions.TYPESCRIPT_DEFINITION_FILE) || stat.isDirectory(), { enumerateDirectories: false });
+
+							let defs = _.map(definitionFiles, def => {
+								return this.getReferenceLine(fromWindowsRelativePathToUnix(path.relative(projectDir, def)));
+							});
+
+							this.$logger.trace(`Adding lines for definition files: ${definitionFiles.join(", ")}`);
+							lines.push(...defs);
+						}
+					});
+			}
+
+			// TODO: Make sure the android17.d.ts and ios.d.ts are added.
+
+			if (lines.length) {
+				this.$logger.trace("Updating reference file with new entries...");
+				this.$fs.writeFile(pathToReferenceFile, lines.join(os.EOL), "utf8").wait();
+			} else {
+				this.$logger.trace(`Could not find any .d.ts files for ${this.$projectConstants.REFERENCES_FILE_NAME} file. Deleting the old file.`);
+				this.$fs.deleteFile(pathToReferenceFile).wait();
+			}
+		}).future<void>()();
+	}
+
+	private isFromTypesRepo(dependency: string): boolean {
+		return !!dependency.match(/^@types\//);
+	}
+
+	private getReferenceLine(pathToReferencedFile: string): string {
+		return `/// <reference path="${pathToReferencedFile}" />`;
+	}
+
+	private get npmExecutableName(): string {
+		if (!this._npmExecutableName) {
+			this._npmExecutableName = "npm";
+
+			if (this.$hostInfo.isWindows) {
+				this._npmExecutableName += ".cmd";
+			}
+		}
+
+		return this._npmExecutableName;
+	}
+
+	private getNpmArguments(command: string, npmArguments?: string[]): string[] {
+		let args: string[] = npmArguments || [];
+		return args.concat([command]);
```

Comment:
Nevermind - those are only dashed arguments and there's no difference between `$ npm install lodash --save` and `$ npm --save install lodash` so ignore me ❎